### PR TITLE
Port to GTK4

### PR DESCRIPTION
**Pull request by treppenwitz03**

### CODE REVIEW

File: src/Views/WifiPage.vala [hunk 1]
```diff
@@ -231,15 +223,20 @@ public class Network.WifiInterface : Network.Widgets.Page {
...
+
+                    if (active_ap.ssid.compare (menu_item.ssid) == 0) {
+                        found = true;
+                        menu_item.active = true;
+                        active_wifi_item = menu_item;
+                        active_wifi_item.state = state;
+                    }
                 }
+
+                child = child.get_next_sibling ();
```
Comment (jeremypw) [inline src/Views/WifiPage.vala:239]: Could test ` && !found` in `while ()` and lose `break`.

File: src/Views/WifiPage.vala [hunk 2]
```diff
@@ -399,31 +398,33 @@ public class Network.WifiInterface : Network.Widgets.Page {
...
-            connected_frame.child = connected_box;
-            connected_frame.show_all ();
+            var connected_listbox = new Gtk.ListBox () {
+                selection_mode = NONE
+            };
+            connected_listbox.add_css_class (Granite.STYLE_CLASS_RICH_LIST);
+            connected_listbox.append (connected_box);
+            connected_listbox.get_first_child ().focusable = false;
+
+            connected_frame.child = connected_listbox;
```
Comment (jeremypw) [inline src/Views/WifiPage.vala:427]: Non-obvious code could use a comment. Styling workaround?

File: src/MainView.vala [hunk 3]
```diff
@@ -301,11 +300,8 @@ public class Network.MainView : Gtk.Box {
     }
 
     private void remove_iface_from_list (Widgets.Page iface) {
-        foreach (unowned var _list_item in get_children ()) {
-            var list_item = (Widgets.DeviceItem)_list_item;
-            if (list_item.page == iface) {
-                device_list.remove (list_item);
-            }
+        while (device_list.get_row_at_index (0) != null) {
+            device_list.remove (device_list.get_row_at_index (0));
```
[the remark this answers is not in the document]
Comment (danirabbit) [reply src/MainView.vala:304]: Thank you for catching that!

**Comment by danirabbit** [comment]
@jeremypw thanks for thorough review! Addressed your comments. FWIW there are OS 8 daily builds available now :) https://builds.elementary.io

**Comment by jeremypw** [comment]
@danirabbit I did install an OS8 daily build but it resulted in my OS7.1 install becoming unbootable for some reason and I had to reinstall OS 7.1.  Is that a known problem?

**Comment by danirabbit** [comment]
@jeremypw :shrug: Probably need to check in installer's tracker

**Comment by jeremypw** [comment]
It seems both OS7.1 and OS8 will boot if OS7.1 is installed second anyway so I can now use OS8 but it has some annoying issues like not waking from suspend at the moment for me. I'll try to test this PR on it though.

**Comment by danirabbit** [comment]
@jeremypw yeah it's still pretty unstable 😅 might be better for a VM or a second computer for now

**Review by jeremypw — APPROVED** [review]
Confirmed this builds and runs on OS8.  Not fully tested but Wired and Wireless work OK. I would not expect functional changes based on the diff.  Noticed that "Advanced Settings" currently does nothing unless "Gnome Network Manager" is installed (which it isn't by default at the moment) but I guess that needs fixing upstream.

**Comment by jeremypw** [comment]
There are a number of terminal critical warnings produced but I haven't regarded them as blocking as functionality seems unaffected (and they may be present in master)

**Pull request merged**